argus/shell.py

```diff
@@ -0,0 +1,227 @@
+                        help="Make sepatate log files for each scenario.")
+    parser.add_argument("-g", "--git_command", default="",
+                        help="The git command that will test the "
+                             "Cloudbase-Init patch.")
+    parser.add_argument("-z", "--zip_patch", default="",
+                        help="The zip patch to use when testing.")
+    parser.add_argument("-a", "--architecture", default="x64",
+                        help="The OS architecture.")
+    return parser
+
+
+def _prepare_enviroment(local, directory):
+    """Prepare the temp enviroment."""
+    os.mkdir(os.path.join(directory, "ci"))
+
+    testr_conf = os.path.join(directory, ".testr.conf")
+    tests = os.path.join(directory, "ci", "tests.py")
+
+    if local:
+        local = os.path.abspath(local)
+        os.link(os.path.join(local, ".testr.conf"), testr_conf)
+        os.link(os.path.join(local, "ci", "tests.py"), tests)
+    else:
+        # Download the necessary items
+        download_argus_resource(".testr.conf", testr_conf)
+        download_argus_resource("ci/tests.py", tests)
+
+    # Create a new repository
+    subprocess.Popen(["testr", "init"], cwd=directory, close_fds=True).wait()
+
+
+def _prepare_config(separate, resources, flavor_ref,
+                    git_command, zip_patch,
+                    directory, image_ref, architecture):
+    """Prepare the argus config file."""
+
+    conf = six.moves.configparser.SafeConfigParser()
+    conf.add_section("argus")
+    conf.add_section("openstack")
+
+    conf.set("argus", "output_directory", os.path.join(directory, "output"))
+    conf.set("argus", "argus_log_file", os.path.join(directory, "argus.log"))
+    conf.set("argus", "git_command", str(git_command))
+    conf.set("argus", "patch_install", str(zip_patch))
+    conf.set("argus", "log_each_scenario", str(separate))
+    conf.set("argus", "arch", str(architecture))
+    conf.set("openstack", "image_ref", str(image_ref))
+
+    if resources:
+        conf.set("argus", "resources", str(resources))
+
+    if flavor_ref:
+        conf.set("openstack", "flavor_ref", str(flavor_ref))
+
+    config_path = os.path.join(directory, "argus.conf")
+    with open(config_path, 'w') as file_handle:
+        conf.write(file_handle)
+
+    return config_path
+
+
+def _start_testr(parallel, tests, directory):
+    """Start running the tests."""
+
+    cmd = ["testr", "run", str(tests)]
+    if parallel:
+        args = ["--parallel", "--concurrency={}".format(parallel)]
+        cmd.extend(args)
+
+    process = subprocess.Popen(cmd, cwd=directory, close_fds=True)
+    return process
+
+
+def main():
+    """The main entry point."""
+    parser = _prepare_argument_parser()
+    args = parser.parse_args(sys.argv[1:])
+    image_name = _get_image_name(args.image_ref)
+
+    base_directory = _get_base_directory(args.directory, image_name)
+
+    print("Starting at {}".format(base_directory))
```

Comment:
When the directory is not specified a temporary one is created and we need to know the name of it.